src/core/Traits/EntityManagerModel.php

```diff
@@ -0,0 +1,306 @@
+<?php
+namespace MapasCulturais\Traits;
+
+use MapasCulturais\App;
+use MapasCulturais\Entity;
+
+trait EntityManagerModel {
+
+    private $entityOpportunity;
+    private $entityOpportunityModel;
+
+    function ALL_generatemodel(){
+        $app = App::i();
+
+        $this->requireAuthentication();
+        $this->entityOpportunity = $this->requestedEntity;
+        $this->entityOpportunityModel = $this->generateModel();
+
+        $this->generateEvaluationMethods();
+        $this->generatePhases();
+        $this->generateMetadata();
+        $this->generateRegistrationFieldsAndFiles();
+        $this->generateSealsRelations();
+
+        $this->entityOpportunityModel->save(true);
+        
+        if($this->isAjax()){
+            $this->json($this->entityOpportunity);
+        }else{
+            $app->redirect($app->request->getReferer());
+        }
+    }
+
+    function ALL_generateopportunity(){
+        $app = App::i();
+
+        $this->requireAuthentication();
+        $this->entityOpportunity = $this->requestedEntity;
+
+        $app->disableAccessControl();
+        $this->entityOpportunityModel = $this->generateOpportunity();
+
+        $this->generateEvaluationMethods();
+        $this->generatePhases();
+        $this->generateMetadata(0, 0);
+        $this->generateRegistrationFieldsAndFiles();
+
+        $this->entityOpportunityModel->save(true);
+       
+        $app->enableAccessControl();
+
+        $this->json($this->entityOpportunityModel); 
+    }
+
+    function GET_findOpportunitiesModels()
+    {
+        $app = App::i();
+        $dataModels = [];
+        
+        $opportunities = $app->em->createQuery("
+            SELECT 
+                op.id
+            FROM
+                MapasCulturais\Entities\Opportunity op
```

Comment:
Recuperar apenas os registros que são modelo, otimizando o processamento. Isso faz muito diferença para o Ministério, pois há milhares de oportunidades cadastradas em escopo nacional.
```suggestion
            SELECT 
                op.id
            FROM
                MapasCulturais\Entities\Opportunity op
            WHERE
                op.isModel = 1
```